src/cli/util.js

```diff
@@ -995,7 +995,14 @@ function normalizeContextArgv(context, keys) {
   const argv = !keys ? context.argv : pick(context.argv, keys);
 
   context.argv = optionsNormalizer.normalizeCliOptions(argv, detailedOptions, {
-    logger: context.logger,
+    logger: {
+      warn(...args) {
+        if (context.logger) {
+          context.logger.error(...args);
+        }
+        process.exit(1);
+      },
+    },
```

Comment:
@evilebottnawi You know what? Turns out, we only need change this from `warn` to `error`. But those tests..., really a mess to fix.